docs/telemetry-perf.md

```diff
@@ -259,6 +259,291 @@ How long it took from when the user stopped pressing a key to when they were sho
         end
 ```
 
+## Amazon Q Chat
+
+### amazonq_chatRoundTrip
+
+Measures sequential response times in Q chat, from user input to message display. Tracks time intervals between key events: editor receiving the message, feature processing, and final message rendering
+
+```mermaid
+    sequenceDiagram
+        participant User
+        participant chat as Chat UI
+        participant vscode as VSCode
+        participant event as Event Recorder
+        participant partner as Partner team code
+        participant telemetry
+
+        User->>chat: Write chat message and press enter
+        chat->>vscode: send message with timestamp
+        vscode->>event: record chatMessageSent/editorReceivedMessage timestamps
+        vscode->>partner: forward chat message
+        partner->>event: record featureReceivedMessage timestamp
+        partner->>partner: call backend/get response
+        partner->>vscode: forward response contents
+        vscode->>chat: display message
+        chat->>vscode: send stop-chat-message-telemetry event
+        vscode->>event: record messageDisplayed timestamp
+        vscode->>telemetry: emit amazonq_chatRoundTrip
+```
+
+### cwsprChatTimeToFirstChunk
+
+The time between when the conversation id was created and when we got back the first usable result
+
+```mermaid
+    sequenceDiagram
+        participant user as User
+        participant chat as Chat UI
+        participant vscode as VSCode
```

Comment:
Between Chat UI, VS Code, and Backend. I wonder if this is obvious to people reading.

Would Chat UI, Extension Backend, Service Backend make more sense? I'm not hardset on these specific values.
Or maybe a legend that explains each participant can help